File: Python/7py 1/cost.py
"""
|--------------------------------------------------------------------------
| Driving School Series #2
|--------------------------------------------------------------------------
|
"""

from math import ceil
def cost(mins):
   return ceil(max(0, mins - 65) / 30 + 3) * 10


def cost(n):
   n -= 5
   if n<60 : return 30
   n -= 60
   return 30 + ceil(n/30) * 10


def cost(mins):
   total = mins
   total -= 60
   price = 30
   while total > 5:
      total -= 30
      price += 10
   return price


def cost(mins):
   return max(0, -(-(mins - 65)//30)) * 10 + 30


def cost(mins):
   return 30 if mins<=60 else 30+10*int((mins-60+24)/30)


print(cost(45),30)
print(cost(63),30)
print(cost(84),40)
print(cost(102),50)
print(cost(273),100)

File: JavaScript/7js 1/cost.js
/*
|--------------------------------------------------------------------------
| Driving School Series #2
|--------------------------------------------------------------------------
|
*/

function cost(mins) {
   const x = Math.floor((mins - 60) / 30) * 10
   const y = mins % 30 > 5 ? 10 : 0;
   return mins - 60 < 0 ? 30 : x + y + 30;
}


function cost(mins) {
   return 30 + (mins > 65 ? Math.ceil((mins - 65) / 30) : 0) * 10
}


function cost(mins) {
   let sum = 30
   mins -= 60
   while (mins > 5) {
      sum += 10
      mins -= 30
   }
   return (sum);
}


function cost(mins) {
   return Math.ceil(Math.max(0, mins - 65) / 30 + 3) * 10;
}


function cost(mins) {
   return mins <= 60 ? 30 : 30 + 10 * (Math.ceil((mins - 60 - 5) / 30))
}


console.log(cost(45), 30);
console.log(cost(63), 30);
console.log(cost(84), 40);
console.log(cost(102), 50);
console.log(cost(273), 100);